### PR TITLE
MINOR: logging improvements

### PR DESCRIPTION
**Pull request by guozhangwang**
This is a manual cherry-pick of https://github.com/apache/kafka/pull/3769 for 0.11.0

### CODE REVIEW

**Comment by guozhangwang** [comment]
@mjsax @dguy @bbejeck please review.

**Comment by dguy** [comment]
@guozhangwang looks like you have checkstyle errors

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java [hunk 1]
```diff
@@ -62,7 +62,7 @@
     }
 
     void addNewTask(final T task) {
-        log.trace("{} add new {} {}", logPrefix, taskTypeName, task.id());
+        log.trace("{} Add newly created {} {} with assigned partitions {}", logPrefix,  taskTypeName, task.id(), task.partitions());
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java:65]: nit: extra space before `taskTypeName`

**Review by dguy — COMMENTED** [review]
There are checkstyle errors but otherwise LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 2]
```diff
@@ -189,8 +189,9 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
                 log.trace("{} pausing partitions: {}", logPrefix, partitions);
                 consumer.pause(partitions);
             } catch (final Throwable t) {
+                log.error("{} Error caught during partition assignment, " +
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:192]: nit: put the error message earlier in the log statement  `Error caught during partition assignment {},`

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 3]
```diff
@@ -220,8 +221,9 @@ public void onPartitionsRevoked(final Collection<TopicPartition> assignment) {
                 // suspend active tasks
                 suspendTasksAndState();
             } catch (final Throwable t) {
+                log.error("{} Error caught during partition revocation, " +
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:224]: nit: same as above

**Comment by bbejeck** [comment]
just two minor comments otherwise LGTM.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 4]
```diff
@@ -1074,23 +1076,19 @@ private void closeNonAssignedSuspendedStandbyTasks() {
...
-                    consumer,
-                    storeChangelogReader,
-                    config,
-                    streamsMetrics,
-                    stateDirectory,
-                    cache,
-                    time,
-                    createProducer(id));
-        } finally {
-            log.trace("{} Created active task {} with assigned partitions {}", logPrefix, id, partitions);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:1092]: Should we keep the log entry (ie, move out of `finally` block) ?

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 5]
```diff
@@ -1154,11 +1152,7 @@ protected StandbyTask createStandbyTask(final TaskId id, final Collection<TopicP
         final ProcessorTopology topology = builder.build(id.topicGroupId);
 
         if (!topology.stateStores().isEmpty()) {
-            try {
-                return new StandbyTask(id, applicationId, partitions, topology, consumer, storeChangelogReader, config, streamsMetrics, stateDirectory);
-            } finally {
-                log.trace("{} Created standby task {} with assigned partitions {}", logPrefix, id, partitions);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:1160]: As above.

**Review by mjsax — COMMENTED** [review]
Two minor comments.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 5 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:1160]: We do not need to, as the returned value would be printed again in the `addNewTask` call (that's also why I added a few more pieces of info there).

**Comment by guozhangwang** [comment]
Merged to 0.11.0 after fixing checkstyle issues (local run passed).

**Pull request closed without merging**